location/views.py

```diff
@@ -66,3 +68,52 @@ def review(request, pk):
             )
             r.save()
     return HttpResponseRedirect(reverse("location", args=(pk,)))
+
+
+@login_required
+def apartment_upload(request):
+    # if this is a POST request we need to process the form data
+    if request.method == "POST":
+        # create a form instance and populate it with data from the request:
+        form = ApartmentUploadForm(request.POST, request.FILES)
+        # check whether it's valid:
+        if form.is_valid():
+            # process the data in form.cleaned_data as required
+            city = form.cleaned_data["city"]
+            state = form.cleaned_data["state"]
+            address = form.cleaned_data["address"]
+            zipcode = form.cleaned_data["zipcode"]
+            rent_price = form.cleaned_data["rent_price"]
+            image = form.cleaned_data["image"]
+            suite_num = form.cleaned_data["suite_num"]
+
+            # TODO use geocoding API to get latitude and longitude
+
+            # create or retrieve an existing location
+            loc = Location.objects.get_or_create(
+                city=city, state=state, address=address, zipcode=zipcode
+            )[
+                0
+            ]  # using get_or_create avoids race condition
+
+            loc.save()
+
+            # create an apartment and link it to that location
+            apt = Apartment.objects.create(
+                suite_num=suite_num, image=image, rent_price=rent_price, location=loc
```

Comment:
should there be some validation on the `suite_num`? It's not a huge issue's down the line if we have multiple Apartments with the same suite No for a particular location.

Reply:
Good idea! Yes, we should check if there are any apartments with the same suite number in the current location. Added some validation for that!